src/dfcx_scrapi/tools/analysis_util.py

```diff
@@ -0,0 +1,162 @@
+"""Util class for performing analysis on DFCX objects and data."""
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import pandas
+import numpy as np
+from dfcx_scrapi.core.intents import Intents
+import google.cloud.dialogflowcx_v3beta1.types as types
+
+# logging config
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(asctime)s %(levelname)-8s %(message)s',
+    datefmt='%Y-%m-%d %H:%M:%S')
+
+class AnalysisUtils():
+    """Utils class for quickly building DFCX webhook payloads."""
+
+    @staticmethod
+    def levenshtein_ratio(s, t):
+        """ levenshtein_ratio_and_distance:
+            Calculates levenshtein distance between two strings.
+        """
+        rows = len(s)+1
+        cols = len(t)+1
+        distance = np.zeros((rows,cols),dtype = int)
+
+        for i in range(1, rows):
+            for k in range(1,cols):
+                distance[i][0] = i
+                distance[0][k] = k
+
+        for col in range(1, cols):
+            for row in range(1, rows):
+                if s[row-1] == t[col-1]:
+                    cost = 0
+                else:
+                    cost = 2
+
+                distance[row][col] =  min(
+                    distance[row-1][col] + 1,
+                    distance[row][col-1] + 1,
+                    distance[row-1][col-1] + cost
+                )
+
+        # Computation of the Levenshtein Distance Ratio
+        ratio = ((len(s)+len(t)) - distance[row][col]) / (len(s)+len(t))
+        return ratio
+
+
+    @staticmethod
+    def calc_tp_distances(
```

Comment:
@Omerside add Type Hinting in the args here, especially for `intent_key` and `intent_comparator` since you are introducing some new terminology that is outside of the typical bounds of the CX paradigm.